docs/csharp/language-reference/keywords/in-parameter-modifier.md

```diff
@@ -37,32 +37,89 @@ class CS0663_Example
 }
 ```
   
-Overloading based on the presence of `in` is allowed, but generates a compiler warning:  
+Overloading based on the presence of `in` is allowed:  
   
 ```csharp
 class InOverloads
 {
-    // Discouraged. Calling SampleMethod(value) is ambiguous.
     public void SampleMethod(in int i) { }
     public void SampleMethod(int i) { }
 }
 ```
 
-Properties or constants may be passed as `in` parameters, because the calling method may not modify their values.
-  
+## Overload resolution rules
+
+You can most easily understand the overload resolution rules for methods with by value vs. `in` arguments like the previous example methods through understanding the motivation for `in` arguments. Defining methodss using `in` parameters is a potential performance optimization. Some `struct` type arguments may be large in size, and when methods are called in tight loops or critical code paths, the cost of copying those structures is critical. Methods declare `in` parameters to specify that arguments may be passed by reference safely because the called method does not modify the state of that argument. Passing those arguments by reference avoids the (potentially) expensive copy. 
+
+Specifying `in` on arguments at the call site is typically optional because the `in` modifier does not allow the value of the argument to be modified. You would explicitly add the `in` modifier at the callsite to ensure the argument is passed by reference, not by value. Explicitly using `in` has two effects:
+
+First, specifying `in` at the call site forces selection of a method defined with a matching `in` argument. Otherwise, when two methods differ only in the presence of `in`, the by value overload is a better match.
+
+Second, specifying `in` declares your intent to pass an argument by reference. The argument used with `in` must represent a location that can be directly referred to. The same general rules for `out` and `ref` arguments apply: You cannot use constants, properties or other expressions. Omitting `in` at the callsite informs the compiler that you will allow the compiler to create a temporary variable to pass by read only reference to the method. Enabling the compiler to create a temporary has several implications:
```

Comment:
technically properties can be ref-returning and then they are lvalues and can be used as "in" arguments.
Perhaps says something like " . . . constants, ordinary properties or other expressions that produce values."